ui/src/app/settings/components/repos-list/repos-list.tsx

```diff
@@ -208,31 +208,34 @@ export class ReposList extends React.Component<
                     url: !sshValues.url && 'Repository URL is required'
                 };
             case ConnectionMethod.HTTPS:
-                const httpsValues = params as NewHTTPSRepoParams;
+                const validURLValues = params as NewHTTPSRepoParams;
                 return {
                     url:
-                        (!httpsValues.url && 'Repository URL is required') ||
-                        (this.credsTemplate && !this.isHTTPSUrl(httpsValues.url) && !httpsValues.enableOCI && 'Not a valid HTTPS URL'),
-                    name: httpsValues.type === 'helm' && !httpsValues.name && 'Name is required',
-                    username: !httpsValues.username && httpsValues.password && 'Username is required if password is given.',
-                    password: !httpsValues.password && httpsValues.username && 'Password is required if username is given.',
+                        (!validURLValues.url && 'Repository URL is required') ||
+                        (this.credsTemplate && !this.isHTTPOrHTTPSUrl(validURLValues.url) && !validURLValues.enableOCI && 'Not a valid HTTP/HTTPS URL'),
+                    name: validURLValues.type === 'helm' && !validURLValues.name && 'Name is required',
+                    username: !validURLValues.username && validURLValues.password && 'Username is required if password is given.',
+                    password: !validURLValues.password && validURLValues.username && 'Password is required if username is given.',
+                    tlsClientCertKey: !validURLValues.tlsClientCertKey && validURLValues.tlsClientCertData && 'TLS client cert key is required if TLS client cert is given.',
                     bearerToken:
-                        (httpsValues.password && httpsValues.bearerToken && 'Either the password or the bearer token must be set, but not both.') ||
```

Comment:
You're just renaming the httpsValues constant to validURLValues, correct?  @todaywasawesome mentioned that this should not be removed.